### DISCUSSION TITLE
Subdirectory Hosting with CKAN in Docker

### DISCUSSION THREAD

**Discussion by rj-podmore** [ckan/ckan#8271] [Q&A]
Hi,

I'm new to CKAN and have been enjoying getting stuck in on my first implementation.

I have a CKAN 2.10 implementation in Docker on an Ubuntu 22.04 server and I have this installed and working fine in localhost on the server.  I'm using NGINX as a reverse proxy server to connect to this localhost implementation and have managed to connect this fine in the config as well.  

The problem I have is that my business requirement is for the CKAN implementation to be hosted as a subdirectory of a main domain - ie.  www.mainurl.com/newckansite .  This is the format it is set to in NGINX outside Docker.

Unfortunately, the problem I have is that CKAN doesn't seem to like this and just returns the CKAN error 404 page minus any styling.  So it appears that it doesn't recognise the subdirectory.  I have tried to alter the site URL in the docker-compose.yml file as suggested on Stack Overflow then docker compose build and then docker compose up.  This makes no difference.

I've done a search on here but nothing came up - does anyone have any ideas please?

**Comment by muhammetcetinyamac** [comment]
Hi, 

You can try to redirect the incoming request by rewriting it as in the block below.

```
location /newckansite/ {
    rewrite ^/newckansite(/.*)$ $1 break;
    proxy_pass <http://www.mainurl.com/>;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
}
```
